### PR TITLE
[Check-in] Update list semantics

### PR DESCRIPTION
**Pull request by brdunfield**
## Description
This PR moves us away from using `dl, dt, and dd` markup towards semantic lists or divs. This should solve our issues of screenreaders reading "definition" repeatedly when accessing our content in these lists.

Note: This PR adds nanoid to a main dependency in package.json. This dependency was already part of our resolution section in package.json, so overall package size should not be impacted.
This package is needed to guarantee unique React keys for appointment list items, as we cannot 100% guarantee the appointment IEN is unique.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#44145

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/components/FieldTemplate.jsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import get from '../../../../utilities/data/get';
 import classNames from 'classnames';
+import get from '../../../../utilities/data/get';
```
Comment (brianseek) [inline src/platform/forms-system/src/js/components/FieldTemplate.jsx:3]: @brdunfield did you mean to make a change to this file?

Comment (brdunfield) [reply src/platform/forms-system/src/js/components/FieldTemplate.jsx]: I did not. Reverted!

File: src/applications/check-in/components/TextInputErrorWrapper.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ const TextInputErrorWrapper = ({ error, children }) =>
 
 TextInputErrorWrapper.propTypes = {
   children: PropTypes.node.isRequired,
-  error: PropTypes.bool.isRequired,
+  error: PropTypes.bool,
```
Comment (acrollet) [inline src/applications/check-in/components/TextInputErrorWrapper.jsx:9]: 🚀

**Review by acrollet — APPROVED** [review]
This looks great!

File: src/applications/check-in/components/AppointmentDisplay/AppointmentListItem.jsx [hunk 3]
```diff
@@ -9,34 +9,32 @@ const AppointmentListItem = props => {
   const { appointment, token, router } = props;
   const appointmentDateTime = new Date(appointment.startTime);
   const { t } = useTranslation();
+  const apptId = `${appointment.stationNo ? appointment.stationNo : ''}${
+    appointment.appointmentIen
+  }`;
```
[the remark this answers is not in the document]
Comment (brianseek) [reply src/applications/check-in/components/AppointmentDisplay/AppointmentListItem.jsx:14]: I had this like this at first and it wasn't working reliably for some reason. I got fed up and went with the longer syntax.

[the remark this answers is not in the document]
Comment (acrollet) [reply src/applications/check-in/components/AppointmentDisplay/AppointmentListItem.jsx:14]: huh, I tried it locally with success but I'd rather have reliability than brevity...

**Pull request merged**